saq/queue/postgres.py

```diff
@@ -0,0 +1,608 @@
+                else:
+                    await job.finish(Status.ABORTED, error="swept")
+        return swept
+
+    async def listen(
+        self,
+        job_keys: Iterable[str],
+        callback: ListenCallback,
+        timeout: float | None = 10,
+    ) -> None:
+        if not job_keys:
+            return
+
+        async with self.pool.connection() as conn:
+            for key in job_keys:
+                await conn.execute(f'LISTEN "{key}"')
+            await conn.commit()
+            gen = conn.notifies(timeout=timeout or None)
+            async for notify in gen:
+                payload = self._load(notify.payload)
+                key = payload["key"]
+                status = Status[payload["status"].upper()]
+                if asyncio.iscoroutinefunction(callback):
+                    stop = await callback(key, status)
+                else:
+                    stop = callback(key, status)
+
+                if stop:
+                    await gen.aclose()
+
+    async def notify(self, job: Job) -> None:
+        payload = self._dump({"key": job.key, "status": job.status})
+        await self._notify(job.key, payload)
+
+    async def _notify(self, channel: str, payload: t.Any) -> None:
+        async with self.pool.connection() as conn:
+            await conn.execute(f"NOTIFY \"{channel}\", '{payload}'")
+
+    async def update(
+        self, job: Job, status: Status | None = None, scheduled: int | None = None
+    ) -> None:
+        if status:
+            job.status = status
+        if scheduled:
+            job.scheduled = scheduled
+        job.touched = now()
+
+        async with self.pool.connection() as conn:
+            await conn.execute(
+                SQL(
+                    """
+                    UPDATE {jobs_table} SET job = %(job)s, status = %(status)s, scheduled = %(scheduled)s
+                    WHERE key = %(key)s
+                    """
+                ).format(jobs_table=self.jobs_table),
+                {
+                    "job": self.serialize(job),
+                    "status": job.status,
+                    "key": job.key,
+                    "scheduled": job.scheduled,
+                },
+            )
+        await self.notify(job)
+
+    async def job(self, job_key: str) -> Job | None:
+        async with self.pool.connection() as conn, conn.cursor() as cursor:
+            await cursor.execute(
+                SQL(
+                    """
+                SELECT job FROM {jobs_table}
+                WHERE key = %(key)s
+                """
+                ).format(jobs_table=self.jobs_table),
+                {"key": job_key},
+            )
+            job = await cursor.fetchone()
+            if job:
+                return self.deserialize(job[0])
+        return None
+
+    async def abort(self, job: Job, error: str, ttl: float = 5) -> None:
+        job.error = error
+        job.status = Status.ABORTING
```

Comment:
where is the job actually aborted?

Reply:
The worker's abort function cancels the task and calls queue.finish_abort(), which updates the status to aborted.